### DISCUSSION TITLE
Could it be? Prescribed mesh movement and using idwarp as mesh solver?

### DISCUSSION THREAD

**Discussion by beinspired** [mdolab/dafoam#519] [General]
Hi, 

I am looking for an alternative to the laplacian solver included in Openfoam for mesh modification and looking at the literature, idwarp seems quite promising due to its robustness. More than for optimization, could idwarp be used for simulate using DAPimpleDyMFoam as solver ( or PimpleDyMFoam) but in my case, prescribing the movement of the wall ? 

Best Regards
Marcelino

**Comment by friedenhe** [comment]
Yes, you can prescribe surface coordinates by calling `setSurfaceCoordinates` and then call `warpMesh` to deform the volume mesh. Check here: https://github.com/mdolab/dafoam/blob/b8d35bbc76955978b430ed8e7495cbb46e45cd2a/dafoam/pyDAFoam.py#L895

**Reply by beinspired** [reply]
Hi Ping He, 

Thanks a lot for answering the question. 
We would like to use a pimpleDyMFoam approach, not specifically an optimization approach. So, every timestep, the mesh should be updated as it is done in the integrated OpenFoam solver. The point is that the official solver works using a laplacian approach what is not very accurate when dealing with high aspect ratio cells in boundary layer. 
Could it be possible to take advantage of the DaFoam/IDwarp development to obtain an accurate dynamicMeshSolver to be used for pimpleDyMFoam for example ( and integrated within the unsteady solver). 

Best Regards
Marcelino

**Reply by friedenhe** [reply]
Yes, it is doable, and we do plan to implement this feature; maybe sometime this year.
